### ISSUE TITLE
Support the removal of default values

### ISSUE THREAD

**Issue by bjonnh** [charleskorn/kaml#1]
It would be nice when
```kotlin
@Serializable
data class Foo(
   val bar: String = "test",
   val foo: String? = null
)
```

to be able to get
```yaml
bar: "test"
```

and not
```yaml
bar: "test"
foo: null
```

Similar to what is possible with Json and Protobuf: https://github.com/Kotlin/kotlinx.serialization/issues/58

**Comment by bjonnh** [comment]
In a similar way, jackson allows for that:
mapper.setSerializationInclusion(Include.NON_NULL);

**Comment by charleskorn** [comment]
Hi @bjonnh, thanks for the feedback. I'd love to add this feature but I don't have a lot of spare time right now, so I might not get to it for a while. 

If you're interested in submitting a PR, I'd be happy to give you some pointers and review any work in progress.

**Comment by charleskorn** [comment]
Resolved with #3.
